extensions/wikia/Discussions/utils/DiscussionsActivator.class.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+use Swagger\Client\ApiException;
+use Swagger\Client\Discussion\Api\SitesApi;
+use Swagger\Client\Discussion\Models\SiteInput;
+use Wikia\DependencyInjection\Injector;
+use Wikia\Service\Swagger\ApiProvider;
+
+class DiscussionsActivator {
+
+    const SERVICE_NAME = 'discussion';
+    const TIMEOUT = 5;
+    const SITE_NAME_MAX_LENGTH = 256;
+
+    private $cityId;
+    private $cityName;
+    private $cityLang;
+
+    private $sitesApi;
+    private $logger;
+
+    public function __construct( $cityId = null, $cityName = null, $cityLang = null ) {
+        $this->cityId = $cityId ?? F::app()->wg->CityId;
+        $this->cityName = $cityName ?? F::app()->wg->Sitename;
+        $this->cityLang = $cityLang ?? F::app()->wg->ContLang->getCode();
```

Comment:
I believe it might be better to explicitly use them as constructor params rather than doing it this way.